# HG changeset patch
# User Rik <rik@octave.org>
# Date 1453258831 28800
#      Tue Jan 19 19:00:31 2016 -0800
# Node ID 47908b818865501db75c42e46263205cb300bf61
# Parent  2bbbbb96f631aada05500e9ad9673a461e0c8785
oct-stream.cc: Add FIXME note about bug #46493.

* oct-stream.cc (octave_scan_1): Add FIXME note about bug #46493.

diff --git a/libinterp/corefcn/oct-stream.cc b/libinterp/corefcn/oct-stream.cc
--- a/libinterp/corefcn/oct-stream.cc
+++ b/libinterp/corefcn/oct-stream.cc
@@ -1147,16 +1147,23 @@ octave_scan_1 (std::istream& is, const s
                       ref = 0;
                   }
                 else
                   {
                     if (c2 == '0' || c2 == '1' || c2 == '2'
                         || c2 == '3' || c2 == '4' || c2 == '5'
                         || c2 == '6' || c2 == '7')
                       is >> std::oct >> ref >> std::dec;
+                    else if (c2 == '8' || c2 == '9')
+                    {
+                      // FIXME: Would like to set error state on octave stream.
+                      // See bug #46493.  But only std::istream is input to fcn
+                      // error ("internal failure to match octal format");
+                      ref = 0;
+                    }
                     else
                       ref = 0;
                   }
               }
             else
               {
                 is.putback (c1);
 
